core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -588,6 +591,14 @@ private String extractContentLinkSpec() {
     }
   }
 
+  static String getReportPath( MasterReport report ) {
+    ResourceManager rm = report != null ? report.getResourceManager() : null;
+    BundleResourceManagerBackend backend = rm != null ? (BundleResourceManagerBackend) rm.getBackend() : null;
```

Comment:
You cannot assume that in 100% of the cases. Some weird person may actually call this action/component with a self-constructed MasterReport instance that has not been loaded from a bundle. Use an "instanceof" check here.

You may find it safer to work directly with the resource-key itself. Have a look at the while-loop in "PentahoKettleTransFormFileProducer#computeFullFilename". Combined with a similar loop in AbstractKettleTransformationProducer#computeFullFilename" testing for java.io.File objects, this is the safest solution to the problem and handles all sort of weird side-cases (like loading a prpt-report contained in a zip or jar file).